firstvoices/backend/tests/test_apis/test_category_api.py

```diff
@@ -104,3 +101,155 @@ def test_detail_404(self):
         wrong_endpoint = self.get_detail_endpoint(self.site.slug, "54321")
         response = self.client.get(wrong_endpoint)
         assert response.status_code == 404
+
+    @pytest.mark.django_db
+    def test_contains_word(self):
+        word_entry = DictionaryEntryFactory(site=self.site)
+        category_word = ParentCategoryFactory(site=self.site)
+        category_word.dictionary_entries.add(word_entry)
+
+        # Testing for WORD flag
+        response = self.client.get(
+            self.get_list_endpoint(
+                self.site.slug, query_kwargs={"contains": TypeOfDictionaryEntry.WORD}
+            )
+        )
+        response_data = json.loads(response.content)
+
+        assert response.status_code == 200
+        assert response_data["count"] > 0
+
+        assert response_data["results"][0] == {
+            "id": str(category_word.id),
+            "title": category_word.title,
+            "description": category_word.description,
+            "children": [],
+        }
+
+        # Testing for PHRASE flag, should not return anything
+        response = self.client.get(
+            self.get_list_endpoint(
+                self.site.slug, query_kwargs={"contains": TypeOfDictionaryEntry.PHRASE}
+            )
+        )
+        response_data = json.loads(response.content)
+
+        assert response.status_code == 200
+        assert response_data["count"] == 0
+
+    @pytest.mark.django_db
+    def test_contains_phrase(self):
+        phrase_entry = DictionaryEntryFactory(
+            site=self.site, type=TypeOfDictionaryEntry.PHRASE
+        )
+        category_phrase = ParentCategoryFactory(site=self.site)
+        category_phrase.dictionary_entries.add(phrase_entry)
+
+        # Testing for PHRASE flag
+        response = self.client.get(
+            self.get_list_endpoint(
+                self.site.slug, query_kwargs={"contains": TypeOfDictionaryEntry.PHRASE}
+            )
+        )
+        response_data = json.loads(response.content)
+
+        assert response.status_code == 200
+        assert response_data["count"] > 0
+
+        assert response_data["results"][0] == {
+            "id": str(category_phrase.id),
+            "title": category_phrase.title,
+            "description": category_phrase.description,
+            "children": [],
+        }
+
+        # Testing for WORD flag, should not return anything
+        response = self.client.get(
+            self.get_list_endpoint(
+                self.site.slug, query_kwargs={"contains": TypeOfDictionaryEntry.WORD}
+            )
+        )
+        response_data = json.loads(response.content)
+
+        assert response.status_code == 200
+        assert response_data["count"] == 0
+
```

Comment:
Please add a test case for what happens on invalid input, like {url}?type=oyster